integration/container/mounts_linux_test.go

```diff
@@ -265,3 +267,36 @@ func TestContainerBindMountNonRecursive(t *testing.T) {
 		poll.WaitOn(t, container.IsSuccessful(ctx, client, c), poll.WithDelay(100*time.Millisecond))
 	}
 }
+
+func TestMountSubPath(t *testing.T) {
+	skip.If(t, testEnv.IsRemoteDaemon)
+
+	defer setupTest(t)()
+
+	ctx := context.Background()
+	d := daemon.New(t)
+	client := d.NewClientT(t)
+
+	d.StartWithBusybox(t)
+	defer d.Stop(t)
+
+	volumeName := t.Name() + "_vol_foo"
+	_, err := client.VolumeCreate(context.Background(), volumetypes.VolumeCreateBody{Name: volumeName})
+	assert.NilError(t, err)
+
+	mnt := mounttypes.Mount{
+		Type:   mounttypes.TypeVolume,
+		Source: volumeName,
+		Target: "/foo",
+	}
+
+	cID := container.Run(t, ctx, client, container.WithMount(mnt), container.WithCmd("sh", "-c", "mkdir /foo/bar; echo xyz > /foo/bar/file"))
+	poll.WaitOn(t,
+		container.IsInState(ctx, client, cID, "exited"),
+		poll.WithDelay(100*time.Millisecond),
+	)
+
+	mnt.VolumeOptions = &mounttypes.VolumeOptions{SubPath: "/bar"}
+	cID = container.Run(t, ctx, client, container.WithMount(mnt), container.WithCmd("sh", "-c", "cat /foo/file"))
+	poll.WaitOn(t, container.IsSuccessful(ctx, client, cID), poll.WithDelay(100*time.Millisecond))
```

Comment:
Wondering; does `container.Run` not fail by itself already? (i.e., do we need the `poll.WaitOn` to check manually?) 🤔

Reply:
Looking at the code (and running a few quick tests), `container.Run` does not fail if the command running in the container exits with non-zero error code, which is what this test uses to make sure the mounting of the sub-dir was done correctly in the second container.